test/ConnectionManagerTest.js

```diff
@@ -100,5 +108,17 @@ describe('ConnectionManager', function () {
                 done(new Error('Client should be retrying!\n' + e));
             }
         });
-    })
+    });
+
+    it('should throw IllegalStateError if there is an incompatible server', function () {
```

Comment:
Isn't this the same test as `gives up connecting after timeout`?

Reply:
No, that test just checks first connection attempt and was passing. Client was hanging at the second attempt. This test checks whether client hangs or not after first attempt.

Reply:
Nevermind, I see the difference. However, I did not like this test as it assumes a certain order of running tests. The test shutdowns the cluster that the others depend on. You can isolate the unresponsive server by starting it at a different port and having client configuration list it.

Reply:
However, client will try `5701` port when it cannot connect. Therefore, I need to run the server on `5701`.

Reply:
Why should it? If I give only a single port in config, will it try 5701 anyway? It may try but I doubt it should.

Reply:
It adds here: https://github.com/hazelcast/hazelcast-nodejs-client/blob/22ecdbf996a35761c63aee0eccd72ef919160d7b/src/connection/DefaultAddressProvider.ts#L39

Reply:
Actually, you are right when there is no address passed on config, it automatically tries to connect to `5071`. When I pass `127.0.0.1:9999`, it won't try to connect to `5071`.